kratos/geometries/coupling_geometry.h

```diff
@@ -337,6 +304,236 @@ template<class TPointType> class CouplingGeometry
+                }
+            }
+            std::vector<double> all_intersections;
+            MergeSpans(all_intersections, intersection_master_spans, intersection_master_spans_on_master_spans);
+
+            IntegrationPointUtilities::CreateIntegrationPoints1D(rIntegrationPoints, all_intersections, rIntegrationInfo.GetNumberOfIntegrationPointsPerSpan(0));
+        }
+    }
+
+    /* @brief This method creates a list of quadrature point geometries
+     *        from a list of integration points.
+     *        Initially, all integration points are transformed to
+     *        quadrature point geometries on the master. Then all points
+     *        are mapped to all slave geometries. Finally, coupling geometries,
+     *        containing quadrature point geometries per integration point on
+     *        master and all slaves are created.
+     *
+     * @param rResultGeometries list of quadrature point geometries.
+     * @param NumberOfShapeFunctionDerivatives the number of evaluated
+     *        derivatives of shape functions at the quadrature point geometries.
+     *
+     * @see quadrature_point_geometry.h
+     */
+    void CreateQuadraturePointGeometries(
+        GeometriesArrayType& rResultGeometries,
+        IndexType NumberOfShapeFunctionDerivatives,
+        const IntegrationPointsArrayType& rIntegrationPoints,
+        IntegrationInfo& rIntegrationInfo) override
+    {
+        const double model_tolerance = 1e-3;
+
+        const SizeType num_integration_points = rIntegrationPoints.size();
+
+        if (rResultGeometries.size() != num_integration_points) {
+            rResultGeometries.resize(num_integration_points);
+        }
+
+        // Create quadrature points on master
+        GeometriesArrayType master_quadrature_points(num_integration_points);
+        mpGeometries[0]->CreateQuadraturePointGeometries(
+            master_quadrature_points,
+            NumberOfShapeFunctionDerivatives,
+            rIntegrationPoints,
+            rIntegrationInfo);
+
+        // Compute vector of location
+        CoordinatesArrayVectorType integration_points_global_coords_vector(num_integration_points);
+        for (SizeType i = 0; i < num_integration_points; ++i) {
+            integration_points_global_coords_vector[i] = master_quadrature_points[i].Center();
+        }
+
+        // First slave
+        IntegrationPointsArrayType integration_points_slave = rIntegrationPoints;
+        CoordinatesArrayType local_slave_coords = ZeroVector(3);
+        CoordinatesArrayType global_slave_coords = ZeroVector(3);
+
+        if (!rIntegrationInfo.Is(IntegrationInfo::DO_NOT_CREATE_TESSELLATION_ON_SLAVE)) {
+            if (this->Dimension() == 1) {
+                CurveTessellation<PointerVector<TPointType>> curve_tesselation;
+                curve_tesselation.Tessellate(*(mpGeometries[1].get()), 1e-2, mpGeometries[1]->PolynomialDegree(0));
```

Comment:
Do you always only have one slave, becaus in CreateQuadrature you have a loop.

Reply:
Yes, that is true.

Create integration points is capable to capture multiple curves. Whereas, up to date, CreateQuadraturePoints is not. This, however can be extended, once needed.